tracee-ebpf/tracee/net_events.go

```diff
@@ -206,3 +176,48 @@ func parseEventMetaData(payloadBytes []byte) (EventMeta, *bytes.Buffer) {
 	return eventMetaData, bytes.NewBuffer(payloadBytes[32:])
 
 }
+
+func netPacketProtocolHandler(buffer *bytes.Buffer, evtMeta EventMeta, ctx procinfo.ProcessCtx, eventName string, bootTime uint64) (external.Event, PacketMeta) {
+	var evt external.Event
+	packet, err := ParseNetPacketMetaData(buffer)
+	if err != nil {
+		return evt, packet
+	}
+	evt = CreateNetEvent(evtMeta, ctx)
+	CreateNetPacketMetaArgs(&evt, packet)
+	evt.EventName = eventName
+	evt.Timestamp = int(evtMeta.TimeStamp + bootTime)
+	return evt, packet
+}
+
+// parsing the PacketMeta struct from bytes.buffer
+func ParseNetPacketMetaData(payload *bytes.Buffer) (PacketMeta, error) {
+	var pktMetaData PacketMeta
+	err := binary.Read(payload, binary.LittleEndian, &pktMetaData)
+	if err != nil {
+		return pktMetaData, err
+	}
+	return pktMetaData, nil
+}
+
+//takes the packet metadata and create argument array with that data
+func createNetPacketMetadataArg(packet PacketMeta) []external.Argument {
+	eventArgs := make([]external.Argument, 0, 0)
+	arg := external.PktMeta{}
+	arg.SrcIP = parseIP(packet.SrcIP)
+	arg.DestIP = parseIP(packet.SrcIP)
+	arg.SrcPort = packet.SrcPort
+	arg.DestPort = packet.DestPort
+	arg.Protocol = packet.Protocol
+	evtArg := external.Argument{
+		ArgMeta: external.ArgMeta{"metadata", "external.PktMeta"},
+		Value:   arg,
+	}
+	eventArgs = append(eventArgs, evtArg)
+	return eventArgs
+}
+
+func CreateNetPacketMetaArgs(event *external.Event, NetPacket PacketMeta) {
+	event.Args = createNetPacketMetadataArg(NetPacket)
+	event.ArgsNum = len(event.Args)
```

Comment:
Now it will be always 1, but after the dns will be it will be more then 1.
Because of that in dans there the `pktMeta` and after that the dns info (requests and responses)